R/generics.R

```diff
@@ -159,52 +177,60 @@ setMethod(
 
 #' Number of lines required to print a value
 #'
+#' @inheritParams open_font_dev
 #' @param x (`ANY`)\cr the object to be printed.
 #' @param colwidths (`numeric`)\cr column widths (if necessary).
 #' @param max_width (`numeric(1)`)\cr width that strings should be wrapped to when
 #'   determining how many lines they require.
+#' @param col_gap (`numeric(1)`)\cr width of gap between columns in number of spaces.
+#'   Only used by methods which must calculate span widths after wrapping.
 #'
 #' @return The number of lines needed to render the object `x`.
 #'
 #' @export
 setGeneric(
   "nlines",
-  function(x, colwidths = NULL, max_width = NULL) standardGeneric("nlines")
+  ## XXX TODO come back and add fontspec default value once not having
+  ## it has found all the disconnection breakages
+  function(x, colwidths = NULL, max_width = NULL, fontspec, col_gap = NULL) standardGeneric("nlines")
 )
 
 ## XXX beware. I think it is dangerous
 #' @export
 #' @rdname nlines
 setMethod(
   "nlines", "list",
-  function(x, colwidths, max_width) {
+  function(x, colwidths, max_width, fontspec, col_gap = NULL) {
     if (length(x) == 0) {
       0L
     } else {
-      sum(unlist(vapply(x, nlines, NA_integer_, colwidths = colwidths, max_width = max_width)))
+      sum(unlist(vapply(x, nlines, NA_integer_, colwidths = colwidths, max_width = max_width, fontspec = fontspec)))
     }
   }
 )
 
 #' @export
 #' @rdname nlines
-setMethod("nlines", "NULL", function(x, colwidths, max_width) 0L)
+setMethod("nlines", "NULL", function(x, colwidths, max_width, fontspec, col_gap = NULL) 0L)
 
 #' @export
 #' @rdname nlines
-setMethod("nlines", "character", function(x, colwidths, max_width) {
+setMethod("nlines", "character", function(x, colwidths, max_width, fontspec, col_gap = NULL) {
+  splstr <- strsplit(x, "\n", fixed = TRUE)
   if (length(x) == 0) {
     return(0L)
-  }
+  } ## else if(is.null(colwidths) && is.null(max_width)) { ## don't need wrapping
+  ##   return(length(splstr))
+  ## }
```

Comment:
colwidths is not used in this code, hence the absence of this consideration right? I do not remember exactly why though

Reply:
this looks like it was wrong anyway, as length(splstr) would be length(x) because of how strsplit works, should be either `lengths(splstr)` (note the s) or if x is guaranteed length 1, it could be `length(splstr[[1]])`. But its commented out anyway so I'll just remove the comment

Reply:
not removed ;)